Elemental 0.87 release Intel 17.0.1 compilation error

A fresh build of Elemental 0.87 (01/05/2017) failed to compile in Intel 17.0.1 due to incompatible declarations in DistMatrix, however the compilation went flawless in GCC 5.3.0.

`[13%]Building CXX object CMakeFiles/El.dir/src/core/DistMatrix/Block/CIRC_CIRC.cpp.o`
In file included from `/home/mselvam/Elemental/version_source2/Elemental/src/core/DistMatrix/Block/CIRC_CIRC.cpp(15):`
`/home/mselvam/Elemental/version_source2/Elemental/src/core/DistMatrix/Block/./setup.hpp(168): error:` `declaration is incompatible with "El::DistMatrix<Ring, El::DistNS::CIRC, El::DistNS::CIRC,` `El::DistWrapNS::BLOCK>::diagType *El::DistMatrix<Ring, El::DistNS::CIRC, El::DistNS::CIRC,` `El::DistWrapNS::BLOCK>::ConstructDiagonal(const El::Grid &, int) const" (declared at line 68 of` `"/home/mselvam/Elemental/version_source2/Elemental/include/El/core/DistMatrix/Block/CIRC_CIRC.hpp")`
  `BDM::ConstructDiagonal`
       ^
 
`compilation aborted for` `/home/mselvam/Elemental/version_source2/Elemental/src/core/DistMatrix/Block/CIRC_CIRC.cpp (code 2)`
`make[2]: *** [CMakeFiles/El.dir/src/core/DistMatrix/Block/CIRC_CIRC.cpp.o] Error 2`

--- a/src/core/DistMatrix/Block/setup.hpp
+++ b/src/core/DistMatrix/Block/setup.hpp
@@ -163,8 +163,7 @@ DistMatrix<T,ROWDIST,COLDIST,BLOCK>* BDM::ConstructTranspose
 { return new DistMatrix<T,ROWDIST,COLDIST,BLOCK>(g,root); }
 
 template<typename T>
-DistMatrix<T,DiagCol<COLDIST,ROWDIST>(),
-             DiagRow<COLDIST,ROWDIST>(),BLOCK>*
+typename BDM::diagType*
 BDM::ConstructDiagonal
 ( const El::Grid& g, int root ) const
 { return new DistMatrix<T,DiagCol<COLDIST,ROWDIST>(),

--- a/src/core/DistMatrix/Element/setup.hpp
+++ b/src/core/DistMatrix/Element/setup.hpp
@@ -147,8 +147,7 @@ DistMatrix<T,ROWDIST,COLDIST>* DM::ConstructTranspose
 { return new DistMatrix<T,ROWDIST,COLDIST>(g,root); }
 
 template<typename T>
-DistMatrix<T,DiagCol<COLDIST,ROWDIST>(),
-             DiagRow<COLDIST,ROWDIST>()>*
+typename DM::diagType*
 DM::ConstructDiagonal
 ( const El::Grid& g, int root ) const
 { return new DistMatrix<T,DiagCol<COLDIST,ROWDIST>(),

